ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/network/DefaultP2PNetwork.java

```diff
@@ -359,8 +359,10 @@ void checkMaintainedConnectionPeers() {
     if (!localNode.isReady()) {
       return;
     }
+    final EnodeURL localEnodeURL = localNode.getPeer().getEnodeURL();
     maintainedPeers
```

Comment:
Is there a good reason why self exists in the maintainedPeers set? Or is that being looked at separately?

Reply:
the way the code is now, at startup, you don't know your own enode at the time the static nodes are added to the maintainedPeers set, so you can't avoid adding your own enode